src/examples/ex12f.f

```diff
@@ -54,7 +54,15 @@ program ex12f
       integer    ent
       integer    nentries, nvalues, stencil_indices(5)
 
-      double precision  values(100), tol
+      double precision tol
+
+#if defined(HYPRE_USING_CUDA)
```

Comment:
It looks like the Fortran standard does not support preprocessor macros, but this approach appears to be a common approach.  @p-costa , maybe you can comment on this?

Reply:
Sure! Yes, because in practice `cpp` marcos are supported by the compilers, typically with a `-cpp` flag, or changing the filename to `.F` (or `.F90`). But I'm not sure if the latter approach is as widely supported.

Reply:
Sounds good, @p-costa.  Thanks!